auth/store.go

```diff
@@ -1070,3 +1073,40 @@ func NewTokenProvider(tokenOpts string, indexWaiter func(uint64) <-chan struct{}
 		return nil, ErrInvalidAuthOpts
 	}
 }
+
+func (as *authStore) WithRoot(ctx context.Context) context.Context {
+	if !as.isAuthEnabled() {
+		return ctx
+	}
+
+	var ctxForAssign context.Context
+	if ts, ok := as.tokenProvider.(*tokenSimple); ok && ts != nil {
+		ctx1 := context.WithValue(ctx, "index", uint64(0))
+		prefix, err := ts.genTokenPrefix()
+		if err != nil {
+			plog.Errorf("failed to generate prefix of internally used token")
+			return ctx
+		}
+		ctxForAssign = context.WithValue(ctx1, "simpleToken", prefix)
+	} else {
```

Comment:
We should have a same mechanism for JWT, right? If you don't have a time I'll follow up the implementation. Also we need to care about CN based auth and it will make the solution complicated... let me think about this.

BTW are you using simple token in your production? It's just a testing feature so I'd like to recommend JWT token for production purposes.

Reply:
We use CN based auth in our production environment.  And in the WithRoot's implement, I have added one line code to override the CN based auth to avoid this problem. 
```
ctx = peer.NewContext(ctx, nil)
```
It may be a little tricky, but in etcd3.2 and earlier etcd version, the LeaseRevoke didn't actually authenticate, so I think it's safe to do this.